CMSgov/cms-carts-seds
# MDCT-203 - Upload/Download/List/Delete Uploads

# Description

A few endpoints around download management,
A couple custom error types so we can actually return real error codes.

There are many uploads to a question, so this went slightly off script of the access patterns doc.  Uploads are saved with the State Name + FileID, where File ID is composed from the question_number, a random number, and other identifying info. AWS File naming conventions are as previously followed in CARTS v2, + the question ID identifier. This should make bringing the old downloads into the new system pretty easy to correct for. AWS File Name itself did not change patterns at all.

## How to test

Make sure you have the .env file updated with the local uploads dynamo db table and bucket name.

Run it, and upload files as a state user. You can download and delete them too. Very fun.

## Dependencies

Update your .env files

## Code authors checklist


## Reviewers Checklist (Two different people)


## Assignee
